crates/cubecl-wgpu/src/runtime.rs

```diff
@@ -96,7 +96,7 @@ pub fn init_existing_device(
     queue: Arc<wgpu::Queue>,
     options: RuntimeOptions,
 ) -> WgpuDevice {
-    let device_id = WgpuDevice::Existing(device.as_ref().global_id());
+    let device_id = WgpuDevice::Existing(ptr::from_ref(device.as_ref()) as usize);
```

Comment:
This seems a bit dangerous! No guarantee your next device doesn''t land on the same memory adress - not unlikely even depending how you allocate stuff.

I think maybe just keeping an AtomicU64 which is incremented might be easier. Do have to be careful not to register something twice but that's already the case I think.

(sorry for the driveby comment - I added this code originally)

Reply:
The code should explain your concerns, it does reallocation on the same address:

> No guarantee your next device doesn't land on the same memory address.

```rust
fn main() {
    let mut x = String::from("Hello, world!");
    let addr_x1 = &x as *const _ as usize;
    drop(x);
    x = String::from("Bonjour, le monde!");
    let addr_x2 = &x as *const _ as usize;
    assert_eq!(addr_x1, addr_x2);
}
```

Reply:
Well, exactly - two different strings and yet the adress is the same. So, in this case - you could feasible create two _different_ wgpu devices and yet have Cube think the are the same WgpuDevice, that's not the good! The id should be unique per device.